### PR TITLE
feat: [CDS-36338]: display Allow simultaneous deployments on the same infrastructure under Infra tab for serverless pipeline

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
##### Summary:
- Checkbox for Allow simultaneous deployments on the same infrastructure is added

##### Jira Links:
https://harness.atlassian.net/browse/CDS-36338

##### Screenshots:
Screenshot 2022-04-14 at 6 57 17 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Comment by chetanvekariya-harness** [comment]
trigger cypress

**Comment by chetanvekariya-harness** [comment]
trigger cypress

File: src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx [hunk 1]
```diff
@@ -93,16 +93,14 @@ export const ServerlessSpecEditable: React.FC<ServerlessSpecEditableProps> = ({
...
-              }
-            : {
-                connectorRef: undefined,
-                stage: value.stage === '' ? undefined : value.stage
-              }
+          const data: Partial<ServerlessInfraTypes> = {
+            connectorRef: undefined,
+            stage: value.stage === '' ? undefined : value.stage,
+            allowSimultaneousDeployments: value.allowSimultaneousDeployments
+          }
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx:100]: This looks good. I would have done this - 
{
            connectorRef: undefined,
            stage: value.stage === '' ? undefined : value.stage,
            allowSimultaneousDeployments: value.allowSimultaneousDeployments,
            ...(hasRegion && value.region === '' ? undefined : value.region)
}

This avoids the need for if and mutating the const variable

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx:100]: Will take care of this in next PR.

**Pull request merged**